src/test/java/org/simulator/sbml/SBMLTestSuiteTest.java

```diff
@@ -159,7 +165,73 @@ public void testModel() throws FileNotFoundException, IOException, XMLStreamExce
                 duration = timepoints[timepoints.length - 1]
                         - timepoints[0];
 
-                if (model.getExtension(CompConstants.shortLabel) == null){
+                if (model.getExtension(CompConstants.shortLabel) != null) {
+
+                    // initialize simulator
+                    CompSimulator compSimulator = null;
+                    boolean errorInCompSimulator = false;
+                    try {
+                        compSimulator = new CompSimulator(sbmlFile);
+                    } catch (Exception e) {
+                        errorInCompSimulator = true;
+                    }
+                    Assert.assertNotNull(compSimulator);
+                    Assert.assertFalse(errorInCompSimulator);
+
+                    //solve
+                    MultiTable solution = null;
+                    boolean errorInSolve = false;
+                    try {
+                        double stepSize = (duration / steps);
+                        solution = compSimulator.solve(stepSize, duration);
+                    } catch (Exception e) {
+                        errorInSolve = true;
+                    }
+                    Assert.assertNotNull(solution);
+                    Assert.assertFalse(errorInSolve);
+
+                    //TODO: Add quality measure to check whether solution meets the correct results
+
+                } else if (model.getExtension(FBCConstants.shortLabel) != null) {
+
+                    FluxBalanceAnalysis solver = null;
+                    boolean errorInFBASimulator = false;
+                    try {
+                        solver = new FluxBalanceAnalysis(org.sbml.jsbml.SBMLReader.read(sbmlFile));
```

Comment:
Just curious: why we use the full canonical name here? Do we bind the corresponding library dynamically?

Reply:
@zakharc, as there is an another import used `import org.sbml.jsbml.xml.stax.SBMLReader` that's why it is using full canonical name.